crates/fuel-core/src/database.rs

```diff
@@ -151,22 +151,19 @@ pub struct Database {
     _drop: Arc<DropResources>,
 }
 
-trait DropFnTrait: FnOnce() + Send + Sync {}
-impl<F> DropFnTrait for F where F: FnOnce() + Send + Sync {}
-type DropFn = Box<dyn DropFnTrait>;
-
-impl fmt::Debug for DropFn {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "DropFn")
-    }
-}
-
-#[derive(Debug, Default)]
+type DropFn = Box<dyn FnOnce() + Send + Sync>;
+#[derive(Default)]
 struct DropResources {
     // move resources into this closure to have them dropped when db drops
     drop: Option<DropFn>,
 }
 
+impl fmt::Debug for DropResources {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "DropResources")
+    }
+}
+
```

Comment:
Is this needed?

Reply:
Yes, because it's used by other structs that require `Debug`:

```rust
#[derive(Clone, Debug)]
pub struct Database {
    data: DataSource,
    // used for RAII
    _drop: Arc<DropResources>,
}
```